DroneInventorySystem/src/main/java/com/digitalojt/web/repository/StockInfoRepository.java

```diff
@@ -0,0 +1,60 @@
+package com.digitalojt.web.repository;
+
+import java.util.List;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import com.digitalojt.web.entity.StockInfo;
+
+/**
+ * 
+ * 部品カテゴリー情報テーブルリポジトリー
+ *
+ * @author lin
+ * 
+ */
+@Repository
+public interface StockInfoRepository extends JpaRepository<StockInfo, Integer> {
+
+	/**
+	 * 論理削除フラグが 0 の在庫情報リストを取得（ID 昇順）
+	 * 
+	 * @param deleteFlag
+	 * @return 在庫情報リスト
+	 */
+    List<StockInfo> findByDeleteFlagOrderByStockIdAsc(int deleteFlag);
+
+    
+    /**
+     * 論理削除フラグが 0 
+     * かつ 部品カテゴリーに合致 
+     * かつ 名称で曖昧検索 
+     * かつ 数量条件にに合致する在庫情報リストを取得（ID 昇順）
+     * 
+     * @param category
+     * @param name
+     * @param amount
+     * @param amountCondition
+     * @param deleteFlag
+     * @return 在庫情報リスト
+     */
+    @Query("SELECT s "
+    		+ "FROM StockInfo s "
+    		+ "WHERE (:categoryId IS NULL OR s.categoryInfo.categoryId = :categoryId) "
+    		+ "AND s.name LIKE %:name% "
```

Comment:
SQLインジェクションを考慮すると下記の方が良いかもです
```suggestion
    		+ "AND (:name IS NULL OR s.name LIKE CONCAT('%', :name, '%')) "
```